File: src/main/java/chapter09/PrepareData.java
package chapter09;

import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

/**
 * create 2021-01-02
 * author zy
 * desc 准备数据（点击流）
 */
public class PrepareData {
    public static int couter = 0;
    public static String[] users = {"张三","李四","王五"};
    public static String[] urls = {"/flink","/spark","intsmaze","sql"};
    public static List<ClickBean> getClickData(){
        List<ClickBean> list = new ArrayList<>();
        list.add(new ClickBean(couter ++,getUser(),new Timestamp(System.currentTimeMillis()),getUrl()));
        list.add(new ClickBean(couter ++,getUser(),new Timestamp(System.currentTimeMillis()),getUrl()));
        list.add(new ClickBean(couter ++,getUser(),new Timestamp(System.currentTimeMillis()),getUrl()));
        list.add(new ClickBean(couter ++,getUser(),new Timestamp(System.currentTimeMillis()),getUrl()));
        list.add(new ClickBean(couter ++,getUser(),new Timestamp(System.currentTimeMillis()),getUrl()));
        list.add(new ClickBean(couter ++,getUser(),new Timestamp(System.currentTimeMillis()),getUrl()));
        list.add(new ClickBean(couter ++,getUser(),new Timestamp(System.currentTimeMillis()),getUrl()));
        list.add(new ClickBean(couter ++,getUser(),new Timestamp(System.currentTimeMillis()),getUrl()));
        list.add(new ClickBean(couter ++,getUser(),new Timestamp(System.currentTimeMillis()),getUrl()));
        return list;
    }

    static String getUser(){
        Random random = new Random();
        int index = random.nextInt(3);
        return users[index];
    }

    static String getUrl(){
        Random random = new Random();
        int index = random.nextInt(4);
        return urls[index];
    }
}

File: src/main/java/chapter06/PrepareData.scala
package chapter06

import java.util

import common.Trade

/**
 * @create 2021-01-30
 * @author zy
 */
object PrepareData {
  def getClicksData(): java.util.List[ClickBean] ={
    val result = new java.util.ArrayList[ClickBean]()
    result.add(new ClickBean(1,"张三","/intsmze","2019-07-28 12:00:00"))
    result.add(new ClickBean(2,"李四","/flink","2019-07-28 12:05:00"))
    result.add(new ClickBean(3,"张三","/stream","2019-07-28 12:08:00"))
    result.add(new ClickBean(4,"李四","/intsmze","2019-07-28 12:10:00"))
    result.add(new ClickBean(5,"王五","/flink","2019-07-28 12:12:00"))
    result
  }

  def getTradeData(): java.util.List[Trade] ={
    val result = new util.ArrayList[Trade]()
    result.add(new Trade("张三",38,"安卓手机","2019-07-28 12:20:00"))
    result.add(new Trade("王五",45,"安卓手机","2019-07-28 12:30:00"))
    result.add(new Trade("张三",18,"台式机","2019-07-28 13:20:00"))
    result.add(new Trade("王五",23,"笔记本","2019-07-28 13:40:00"))
    result
  }
}
